### PR TITLE
Task2. DOM-JS

### PR DESCRIPTION
**Pull request by Nataliia-Kholodkova**
[Demo](https://ua-girl.github.io/Booking-KottansHW/)
[CodeBase](https://github.com/UA-girl/Booking-KottansHW)

Please, review. The data in the table below the form updates each time user selects/updates date, time, city or selects/unselects seats.

### CODE REVIEW

File: submissions/UA-girl/Task-JS-DOM/js/main.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
+(function () {
```
Comment (A-Ostrovnyy) [inline submissions/UA-girl/Task-JS-DOM/js/main.js]: No need to wrap all your code by IIFE.

File: submissions/UA-girl/Task-JS-DOM/js/main.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  const form = document.querySelector('.booking-form');
+  const table = document.querySelector('.order-table');
+  let order = {
+    city: 'Київ',
+    date: '2021-01-30',
+    time: '10:00',
+    place: [],
+    total: 0,
+
+    dateString: function () {
```
Comment (A-Ostrovnyy) [inline submissions/UA-girl/Task-JS-DOM/js/main.js]: Good practice function name must contain/start verb.
The current name is too generic.

File: submissions/UA-girl/Task-JS-DOM/js/main.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+  function inputHandler(e) {
+    e.preventDefault();
+    const { name, value } = e.target;
+    let newValue, newTotal;
+    switch (name) {
+      case 'place':
+        order['place'] = Array.from(
+          form.querySelectorAll('input[type="checkbox"]:checked')
+        );
```
Comment (A-Ostrovnyy) [inline submissions/UA-girl/Task-JS-DOM/js/main.js]: Not a good practice. You already have NodeList with 'forEach' method, no need for extra transformation.

File: submissions/UA-girl/Task-JS-DOM/js/main.js [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    totalString: function () {
+      this.total = 0;
+      this.place.forEach((p) => {
+        this.total += +p.dataset.price;
+      });
+      return `${this.total}грн.`;
+    },
+  };
+
+  function updateTable(name, value) {
```
Comment (A-Ostrovnyy) [inline submissions/UA-girl/Task-JS-DOM/js/main.js]: Same as at line 13

File: submissions/UA-girl/Task-JS-DOM/js/main.js [hunk 5]
```diff
@@ -0,0 +1,77 @@
+(function () {
+  const menu = document.querySelector('.navigation-list');
+  const menuButton = document.querySelector('.navigation-button');
+  const form = document.querySelector('.booking-form');
+  const table = document.querySelector('.order-table');
+  let order = {
```
Comment (A-Ostrovnyy) [inline submissions/UA-girl/Task-JS-DOM/js/main.js]: Why you don't use class for this?

File: submissions/UA-girl/Task-JS-DOM/js/main.js [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+
+  function updateTable(name, value) {
+    table.querySelector(`.${name}`).textContent = value;
+  }
+
+  function inputHandler(e) {
+    e.preventDefault();
+    const { name, value } = e.target;
+    let newValue, newTotal;
+    switch (name) {
```
Comment (A-Ostrovnyy) [inline submissions/UA-girl/Task-JS-DOM/js/main.js]: Switch for two cases is overkill. It's fine but simple `if else` will be enough.

**Review by A-Ostrovnyy — CHANGES_REQUESTED** [review]
Good work!

File: submissions/UA-girl/Task-JS-DOM/js/main.js (hunk 1 above)
Comment (Nataliia-Kholodkova) [reply submissions/UA-girl/Task-JS-DOM/js/main.js]: I wanted to "simulate" a moduleness ))) and hide variables from global scope.

**Comment by Nataliia-Kholodkova** [comment]
Thank you for advices ) Changed a little bit a logic - maybe, this way it would be better?

File: submissions/UA-girl/Task-JS-DOM/js/main.js [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+        this.classList.remove('navigation-button__open');
+        this.classList.add('navigation-button__close');
+    } else {
+        this.classList.add('navigation-button__open');
+        this.classList.remove('navigation-button__close');
+    }
+    menu.classList.toggle('navigation-list__active');
+});
+
+window.onload = function () {
```
Comment (A-Ostrovnyy) [inline submissions/UA-girl/Task-JS-DOM/js/main.js]: Avoid adding event listeners using `onevent` attribute use `addEventListener` for this.

**Review by A-Ostrovnyy — CHANGES_REQUESTED** [review]
Good work!

**Review by A-Ostrovnyy — APPROVED** [review]
Well done!

**Comment by A-Ostrovnyy** [comment]
@UA-girl please resolve conflicts and let me know, after that, I will merge your work)

**Comment by Nataliia-Kholodkova** [comment]
Oh, but this is not my folder (( and have not any folders on my machine via this branch except my own. Can you suggest me, how can I resolve? maybe, make pull from this branch? To get this folder on my machine? This is first time I get such problem ((

**Comment by A-Ostrovnyy** [comment]
Switch to branch `main` then pull all changes to your local branch `main` and then merge `main` branch into your task branch (`UA-girl/js-dom-task`)

**Comment by Nataliia-Kholodkova** [comment]
Alexandr, I've made it! ))) Thank you for advise - it helped )

**Pull request merged**